### PR TITLE
Feat/output pop 2

### PR DESCRIPTION
**Pull request by htopazian**
- changing prev / inc measures to numerator and denominator -> n population in age group & n cases
- updating documentation
- updating vignettes

### CODE REVIEW

File: R/render.R [hunk 1]
```diff
@@ -38,53 +50,62 @@ create_prevelance_renderer <- function(
...
+      )
+      renderer$render(paste0('n_', lower, '_', upper), p, timestep) 
+      p <- count_of_detected( 
         detected,
         age,
         lower,
         upper
       )
-      renderer$render(paste0('pv_', lower, '_', upper), p, timestep)
+      renderer$render(paste0('n_', 'detect_', lower, '_', upper), p, timestep) # NEW
```
Comment (giovannic) [inline R/render.R:66]:
```suggestion
      renderer$render(paste0('n_detect_', lower, '_', upper), p, timestep)
```

File: R/render.R [hunk 2]
```diff
@@ -38,53 +50,62 @@ create_prevelance_renderer <- function(
...
       upper <- parameters$severe_prevalence_rendering_max_ages[[i]]
-      p <- epi_from_subpopulation(
+      p <- count_of_detected(
         severe,
         age,
         lower,
         upper
       )
-      renderer$render(paste0('pv_severe_', lower, '_', upper), p, timestep)
+      renderer$render(paste0('n_', 'severe_', lower, '_', upper), p, timestep)
```
Comment (giovannic) [inline R/render.R:77]:
```suggestion
      renderer$render(paste0('n_severe_', lower, '_', upper), p, timestep)
```

File: R/render.R [hunk 3]
```diff
@@ -38,53 +50,62 @@ create_prevelance_renderer <- function(
...
       upper <- parameters$incidence_rendering_max_ages[[i]]
-      p <- epi_from_subpopulation(
+      p <- count_of_detected( 
         target,
         age,
         lower,
         upper
       )
-      renderer$render(paste0('inc_', lower, '_', upper), p, timestep)
+      renderer$render(paste0('n_', 'inc_', lower, '_', upper), p, timestep)
```
Comment (giovannic) [inline R/render.R:97]:
```suggestion
      renderer$render(paste0('n_inc_', lower, '_', upper), p, timestep)
```

File: R/render.R [hunk 4]
```diff
@@ -38,53 +50,62 @@ create_prevelance_renderer <- function(
...
       upper <- parameters$severe_incidence_rendering_max_ages[[i]]
-      p <- epi_from_subpopulation(
+      p <- count_of_detected(
         severe$copy()$and(target),
         age,
         lower,
         upper
       )
-      renderer$render(paste0('inc_severe_', lower, '_', upper), p, timestep)
+      renderer$render(paste0('n_', 'inc_', 'severe_', lower, '_', upper), p, timestep)
```
Comment (giovannic) [inline R/render.R:108]:
```suggestion
      renderer$render(paste0('n_inc_', 'severe_', lower, '_', upper), p, timestep)
```

File: R/render.R [hunk 5]
```diff
@@ -95,13 +116,13 @@ create_clinical_incidence_renderer <- function(birth, parameters, renderer) {
...
       upper <- parameters$clinical_incidence_rendering_max_ages[[i]]
-      p <- epi_from_subpopulation(
+      p <- count_of_detected( 
         target,
         age,
         lower,
         upper
       )
-      renderer$render(paste0('clin_inc_', lower, '_', upper), p, timestep)
+      renderer$render(paste0('n_', 'inc_', 'clinical_', lower, '_', upper), p, timestep)
```
Comment (giovannic) [inline R/render.R:125]:
```suggestion
      renderer$render(paste0('n_inc_', 'clinical_', lower, '_', upper), p, timestep)
```

File: R/render.R [hunk 6]
```diff
@@ -1,15 +1,28 @@
...
+count_of_subpopulation <- function(target, age, lower, upper) {
   in_range <- individual::Bitset$new(target$max_size)$insert(
     which((age >= lower) & (age <= upper))
   )
+  n_in_range <- in_range$size()
+  if (n_in_range == 0) {
+    return(0)
+  }
+  n_in_range
+}
```
Comment (giovannic) [inline R/render.R:10]: Do we need `target` anymore in this function?

File: R/render.R [hunk 7]
```diff
@@ -1,15 +1,28 @@
...
 
+count_of_detected <- function(target, age, lower, upper) {
+  in_range <- individual::Bitset$new(target$max_size)$insert(
+    which((age >= lower) & (age <= upper))
+  )
```
Comment (giovannic) [inline R/render.R:15]: We calculate this twice every timestep now. Once for `count_of_subpopulation` and once for `count_of_detected`. Is there an easy way to calculate it once?

**Review by giovannic — CHANGES_REQUESTED** [review]
Amazing, thanks for this! The implementation and the documentation is sooo clear :)

I've suggested some simplifying changes and made some comments inline.

**Comment by giovannic** [comment]
Also, if you want to fix the github tests for this branch are out of date. You can update them with this commit from our EPI branch: `1d3acf8`

A quick way to pull that change over to this branch is with:

`git cherry-pick 1d3acf8`

File: R/render.R (hunk 6 above)
Comment (htopazian) [reply R/render.R:10]: I think so? so we can calculate the max size of the group in which the detected people come from?

File: R/render.R (hunk 7 above)
Comment (htopazian) [reply R/render.R:15]: I'm trying to combine these into one function which outputs a list with two results (numerator, denominator). Is this what you had in mind? It is failing on the tests so far, "attempt to apply non-function", but could be cool if we get it to work.

count_of_detected <- function(target, age, lower, upper){
  in_range <- individual::Bitset$new(target$max_size)$insert(
    which((age >= lower) & (age <= upper))
  )
  
  denominator <- in_range$size()
  if (denominator == 0) {
    return(0)
  }  
  
  numerator <- in_range$and(target)$size()
  
  result <- c(numerator, denominator)

  result
}

p <- count_of_detected( 
        detected,
        age,
        lower,
        upper
      )
      renderer$render(paste0('n_detect_', lower, '_', upper), p[[1]], timestep)

**Review by giovannic — APPROVED** [review]
🍾 Woop

**Pull request merged**